lib/AST/ASTPrinter.cpp

```diff
@@ -859,11 +859,16 @@ class PrintAST : public ASTVisitor<PrintAST> {
 
       // Get the innermost nominal type context.
       DeclContext *DC;
-      if (isa<NominalTypeDecl>(Current) || isa<ExtensionDecl>(Current))
+      if (isa<NominalTypeDecl>(Current))
         DC = Current->getInnermostDeclContext();
+      else if (isa<ExtensionDecl>(Current))
+        DC = Current->getInnermostDeclContext()->
+          getAsNominalTypeOrNominalTypeExtensionContext();
       else
         DC = Current->getDeclContext();
 
+      assert(DC->isTypeContext());
```

Comment:
Here is the code in getContextSubstitutions():

```
  auto *ownerNominal = dc->getAsNominalTypeOrNominalTypeExtensionContext();
  if (auto *ownerClass = dyn_cast<ClassDecl>(ownerNominal))
    baseTy = baseTy->getSuperclassForDecl(ownerClass, resolver);

  assert(ownerNominal == baseTy->getAnyNominal());
```

Note that regardless of whether you pass in the ExtensionDecl as 'dc' or the NominalTypeDecl, ownerNominal is the same value.